src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/process/DefaultProcessRunner.java

```diff
@@ -34,31 +34,51 @@
  */
 public class DefaultProcessRunner implements ProcessRunner {
   private final boolean async;
-  private final List<ProcessOutputLineListener> stdOutLineListeners;
-  private final List<ProcessOutputLineListener> stdErrLineListeners;
+  private final List<ProcessOutputLineListener> stdOutLineListeners = new ArrayList<>();
+  private final List<ProcessOutputLineListener> stdErrLineListeners = new ArrayList<>();
   private final List<ProcessExitListener> exitListeners;
   private final List<ProcessStartListener> startListeners;
+  private final boolean inheritProcessOutput;
 
   private Map<String, String> environment;
 
   /**
-   * @param async               Whether to run commands asynchronously
-   * @param stdOutLineListeners Client consumers of process standard output. If empty, output will
-   *                            be inherited by parent process.
-   * @param stdErrLineListeners Client consumers of process error output. If empty, output will be
-   *                            inherited by parent process.
-   * @param exitListeners       Client consumers of process onExit event.
-   * @param startListeners      Client consumers of process onStart event.
+   * Base constructor.
+   *
+   * @param async           whether to run commands asynchronously
+   * @param exitListeners   client consumers of process onExit event
+   * @param startListeners  client consumers of process onStart event
    */
-  public DefaultProcessRunner(boolean async, List<ProcessOutputLineListener> stdOutLineListeners,
-                              List<ProcessOutputLineListener> stdErrLineListeners,
-                              List<ProcessExitListener> exitListeners,
-                              List<ProcessStartListener> startListeners) {
+  public DefaultProcessRunner(boolean async,
+                               List<ProcessExitListener> exitListeners,
+                               List<ProcessStartListener> startListeners,
+                               boolean inheritProcessOutput) {
     this.async = async;
-    this.stdOutLineListeners = stdOutLineListeners;
-    this.stdErrLineListeners = stdErrLineListeners;
     this.exitListeners = exitListeners;
     this.startListeners = startListeners;
+    this.inheritProcessOutput = inheritProcessOutput;
+  }
+
+  /**
+   * Constructor that attaches output listeners to a process. It assumes the generated subprocess
+   * does not inherit stdout/stderr.
+   *
+   * @param async                whether to run commands asynchronously
+   * @param exitListeners        client consumers of process onExit event
+   * @param startListeners       client consumers of process onStart event
+   * @param stdOutLineListeners  client consumers of process standard output. If empty, output will
```

Comment:
is this comment true?  If you pass null or empty here isn't ineritProcessOutput still false?